UnityProject/Assets/Scripts/Managers/GameManager.cs

```diff
@@ -754,18 +761,33 @@ public void RestartRound()
 		StartCoroutine(ServerRoundRestart());
 	}
 
-	IEnumerator ServerRoundRestart()
+	private float GetMemeoryUsagePrecentage()
 	{
-		Logger.Log("Server restarting round now.", Category.Round);
-		Chat.AddGameWideSystemMsgToChat("<b>The round is now restarting...</b>");
+		return (Profiler.GetTotalAllocatedMemoryLong() / 1048576) / SystemInfo.systemMemorySize * 100;
+	}
 
-		// Notify all clients that the round has ended
-		EventManager.Broadcast(Event.RoundEnded, true);
+	IEnumerator ServerRoundRestart()
+	{
+		string[] args = Environment.GetCommandLineArgs();
+		if ((ServerShutsDownOnRoundEnd == false || args.Contains("-NoReboot"))
+		    && (ServerAverageFPS >= 45 || GetMemeoryUsagePrecentage() <= 75f) || args.Contains("-AlwaysReboot"))
```

Comment:
surely this would trigger if there's a lot of people?

Reply:
>Surely this would trigger if there's a lot of people

Sounds like performance issue that needs to be delt with; and this would ensure that everything is cleaned up properly so that in the next round those large number of players don't crash the server.

Reply:
Wait a minute, its testing if server FPS is higher than 45  surely it should be checking if there is less than 45?

Reply:
Read the if statement again; it's checking if we don't want to reboot

Reply:
well, when you have to inverse GetMemeoryUsagePrecentage?

Reply:
if the server FPS is greater 45 then restart the round
if the memory percentage is lower than 75 then restart the round
if -AlwaysReboot is set to true then restart the round (i fixed that)
if any of these are false; reboot the server.